src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1494,7 +1526,7 @@ public override BoundNode VisitCall(BoundCall node)
         {
             if (node.Method.MethodKind == MethodKind.LocalFunction)
             {
-                CheckAssigned(node.Method, node.Syntax);
+                CheckAssigned(node.Method.OriginalDefinition, node.Syntax);
```

Comment:
Because of a rather nasty subtle bug that took me a bit to figure out - if it's generic, then we mark the instantiated form as used, which doesn't equal the original definition's unconstructed form, so things get confused (e.g. the local function is marked as not used, and I assume definite assignment screws up, etc).

I'm not sure what "unrelated" things could be affected, as I don't think any other locals can be generic.